packages/gen-ai-hub/src/utils/deployment-cache.ts

```diff
@@ -51,20 +52,75 @@ function createDeploymentCache(cache: Cache<Deployment>) {
      * @param deployments - Deployments to retrieve the IDs and models from.
      */
     setAll: (
-      opts: Omit<DeploymentResolutionOptions, 'model'>,
+      opts: DeploymentResolutionOptions,
       deployments: AiDeployment[]
     ): void => {
-      // go backwards to cache the first deployment ID for each model
-      [...deployments].reverse().forEach(deployment => {
-        cache.set(getCacheKey({ ...opts, model: extractModel(deployment) }), {
-          entry: transformDeploymentForCache(deployment)
+      const cacheEntries = [...deployments].map(deployment =>
```

Comment:
@MatKuhr how do you like that?

my 2 cents: I think it is semantically better, however more error prone.

Reply:
Ok, after multiple tries and still failing tests, I am reverting this. Feel free to check the commit, if you are interested.